Services/Administration/OJS.Services.Administration.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -19,9 +20,25 @@ public class ContestsBusinessService : IContestsBusinessService
         this.userProvider = userProvider;
     }
 
-    public async Task<bool> UserHasContestPermissions(int contestId, string? userId, bool isUserAdmin)
-        => !string.IsNullOrWhiteSpace(userId) &&
-           (isUserAdmin || await this.contestsData.IsUserLecturerInByContestAndUser(contestId, userId));
+    public async Task<bool> UserHasContestPermissions(
+        int contestId,
+        string? userId,
+        bool isUserAdmin)
+    {
+        if (!string.IsNullOrWhiteSpace(userId) && isUserAdmin)
```

Comment:
If the `userId` is null, we should return `false` immediately.

Reply:
I had added other validation, but it ended up redundant, this is why i updated the method. I'll revert this as currently the behavior really is the same.